File: Interview Problems/Parking Lot/Java Code/domain/Vehicle.java
package domain;

import java.util.UUID;

public class Vehicle {

    public enum VehicleType {
        BIKE, CAR, TRUCK, EV
    }

    private final UUID id;
    private final String licensePlate;
    private final VehicleType vehicleType;

    public Vehicle(String licensePlate, VehicleType vehicleType) {
        this.id = UUID.randomUUID();
        this.licensePlate = licensePlate;
        this.vehicleType = vehicleType;
    }

    public UUID getId() { return id; }
    public String getLicensePlate() { return licensePlate; }
    public VehicleType getVehicleType() { return vehicleType; }

    @Override
    public String toString() {
        return "Vehicle{id=" + id + ", plate=" + licensePlate + ", type=" + vehicleType + "}";
    }
}

File: Interview Problems/Parking Lot/Python Code/domain/vehicle.py
import uuid
from enum import Enum, auto

class VehicleType(Enum):
    BIKE = auto()
    CAR = auto()
    TRUCK = auto()
    EV = auto()

class Vehicle:
    def __init__(self, license_plate: str, vehicle_type: VehicleType):
        self.id = uuid.uuid4()
        self.license_plate = license_plate
        self.vehicle_type = vehicle_type

    def __repr__(self):
        return f"Vehicle(id={self.id}, plate='{self.license_plate}', type={self.vehicle_type.name})"
